Core/Core/Database/CoreData/Models/Assignment.swift

```diff
@@ -91,8 +91,9 @@ extension Assignment {
 
         if let apiRubrics = item.rubric, apiRubrics.count > 0 {
             self.rubric = Set<Rubric>()
-            for var r in apiRubrics {
+            for (index, var r) in apiRubrics.enumerated() {
                 r.assignmentID = item.id.value
+                r.position = index
```

Comment:
I think what you are suggesting makes a lot of sense.  One potential issue i see this in this particular case is that I follow the other `save` methods where you search for an existing model in the database or insert a new one ( I don't think that is ever used though because of the way we delete on collections ) but the predicate requires the assignmentID to find an existing one.
```
        let model: Rubric = context.fetch(predicate).first ?? context.insert()
```
we could not do a fetch???

Reply:
I think either way is fine. It makes sense since the predicate relies on it, that it needs to be added to API struct.